src/Request.js

```diff
@@ -306,13 +307,15 @@ export default class Request {
                 new_path += '?' + data;
             }
         }
-        else if (this.auto_format) {
-            payload = (!this.headers['Content-Type'] && this.data)
-                ? Request.stringify(this.data)
-                : JSON.stringify(this.data);
-        }
-        else {
-            payload = this.data;
+        else if (!this._formatted_payload) {
+            if (this.auto_format) {
+                this._formatted_payload = (!this.headers['Content-Type'] && this.data)
```

Comment:
We discovered that there is a bug with this condition. See this consequence table:

| content-type   | data           |    | parsing |
|----------------|----------------|----|---------|
| urlencoded     | exists         | -> | JSON    |
| urlencoded     | null/undefined | -> | JSON    |
| json           | exists         | -> | JSON    |
| json           | null/undefined | -> | JSON    |
| null/undefined | exists         | -> | Request |
| null/undefined | null/undefined | -> | JSON    |

A bug would exist if content-type is `urlencoded` and and the data exists. Normally, this is fine since we expect the provided data to be urlencoded string already when the user set the content type to `urlencoded`, but this causes an issue for cases like this:

```js
const request = cudl.post
    .to('http://localhost')
    .send({a: 1})
    .then((err) => { // sends payload `a=1`

        // update body and resend
        request.send({a: 2})
            .then((err) => {
                // sends payload `{"a":2}`
            });
    });
```

@pandafox should we still handle cases like this?

Reply:
Doesn't this mean that we should have set the Content-Type as well when the data is being formatted? (since we're checking it for null), a better change may be also setting Content-Type to urlencoded when using Request.stringify?

Either way, please
1. add the table as a comment
1. add in specific test cases that would fail in old condition and pass in new condition.